### PR TITLE
Allow fabric sync example to accept KeepActive Command

### PR DESCRIPTION
**Pull request by tehampson**
This PR does the following three things:
1. Allows fabric-bridge to receive `KeepActive` Command on dynamic bridged node endpoint
2. If device dynamic endpoint is bridging and an ICD device, it will marshal over the parameters from the `KeepActive` command to fabric-admin
3. Sets up marshaling `StayActiveResponse` response when received to back to fabric-bridge so that is can Log and ActiveChanged event.

In a follow up PR we will work on converting a `KeepActive` command received in fabric-admin in item #2 to `StayActiveRequest` when the ICD device is currently active and call the plumbing from item #3

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: examples/fabric-bridge-app/linux/RpcClient.cpp [hunk 1]
```diff
@@ -143,3 +161,22 @@ OpenCommissioningWindow(chip::Controller::CommissioningWindowVerifierParams para
...
+    // function and the call will complete.
+    auto call = fabricAdminClient.KeepActive(params, RpcCompletedWithEmptyResponse);
+
+    if (!call.active())
+    {
+        // The RPC call was not sent. This could occur due to, for example, an invalid channel ID. Handle if necessary.
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    return WaitForResponse(call);
```
Comment (tehampson) [inline examples/fabric-bridge-app/linux/RpcClient.cpp:181]: Arguably we don't have to wait here. There isn't really a benefit to doing so.

File: examples/fabric-admin/rpc/RpcClient.h [hunk 2]
```diff
@@ -61,3 +61,15 @@ CHIP_ERROR AddSynchronizedDevice(chip::NodeId nodeId);
  * - CHIP_ERROR_INTERNAL: An internal error occurred while activating the RPC call.
  */
 CHIP_ERROR RemoveSynchronizedDevice(chip::NodeId nodeId);
+
+/**
+ * @brief Received StayActiveResponse on behalf of client that previously called KeepActive
+ *
+ * @param nodeId The Node ID of the device we recieved a StayActiveResponse.
+ * @param promisedActiveDuration the computed duration (in milliseconds) that the ICD intends to stay active for.
```
Comment (andy31415) [inline examples/fabric-admin/rpc/RpcClient.h:69]: let's name this promisedActiveDurationMs

File: examples/fabric-admin/rpc/RpcServer.cpp [hunk 3]
```diff
@@ -65,6 +65,18 @@ class FabricAdmin final : public rpc::FabricAdmin
 
         return pw::OkStatus();
     }
+
+    pw::Status KeepActive(const chip_rpc_KeepActiveParameters & request, pw_protobuf_Empty & response) override
+    {
+        ChipLogProgress(NotSpecified, "Received KeepActive request: 0x%lx, %u", request.node_id, request.stay_active_duration);
+        // TODO(#33221): When we get this command hopefully we are already registers with an ICD device to be
```
Comment (andy31415) [inline examples/fabric-admin/rpc/RpcServer.cpp:72]:
```suggestion
        // TODO(#33221): When we get this command hopefully we are already registered with an ICD device to be
```

**Review by andy31415 — APPROVED** [review]

File: examples/common/pigweed/protos/fabric_admin_service.proto [hunk 4]
```diff
@@ -14,11 +14,17 @@ message DeviceCommissioningWindowInfo {
   bytes verifier = 6;
 }
 
+message KeepActiveParameters {
+  uint64 node_id = 1;
+  uint32 stay_active_duration = 2;
```
Comment (andy31415) [inline examples/common/pigweed/protos/fabric_admin_service.proto:19]: maybe stay_active_duration_ms here as well for clarity.

**Pull request merged**